src/mds/MDSRank.cc

```diff
@@ -2240,7 +2240,7 @@ int MDSRank::_command_flush_journal(std::stringstream *ss)
 {
   assert(ss != NULL);
 
-  Mutex::Locker l(mds_lock);
+  mds_lock.Lock();
```

Comment:
@batrick I don't know if this change is really appropriate but without it the MDS crashes after the executing the command at this line [1] since it expects MDS to be locked.

[1] https://github.com/ceph/ceph/blob/master/src/mds/MDSDaemon.cc#L1143

Reply:
I'd suggest moving `Mutex::Locker l(mds_lock)` to surround the call to `_command_flush_journal` in `command_flush_journal`:

https://github.com/ceph/ceph/blob/189192b2f60429e95ff798314e71c4252cc6bf7f/src/mds/MDSRank.cc#L2225

to be

````
{
  Mutex::Locker l(mds_lock);
  const int r = _command_flush_journal(&ss);
}
````

Then comment `MDSrank::_command_flush_journal` that it expects `mds_lock` to be locked.